File: benchmarks/landline_latency/echo.py
from orchestratelang import landline


class PyEcho(landline.Serverlet):
    def number(self, n: int) -> int:
        return n

    def text(self, s: str) -> str:
        return s

    def numbers(self, items: list[int]) -> list[int]:
        return items


landline.serve(PyEcho)

File: benchmarks/landline_latency/echo.ts
// The same echo handlers as echo.py, for the TypeScript landline rung: `int` crosses as
// `bigint`, `string` as `string`, and `int[]` as `bigint[]`.
export default class TsEcho {
    number(n: bigint): bigint {
        return n;
    }
    text(s: string): string {
        return s;
    }
    numbers(items: bigint[]): bigint[] {
        return items;
    }
}
